pkg/expr/expr.go

```diff
@@ -390,15 +391,51 @@ func Parse(src string) (ex *Expression, err error) {
 
 // Evaluator for a c-like expression language.
 type cexl struct {
-	//TODO add ast cache
+	lock      sync.RWMutex
+	exprCache map[string]*Expression
```

Comment:
Good catch. Thanks. I have updated the PR to use https://github.com/hashicorp/golang-lru library. The finite size I have currently made as 1024 as a default value during server init. However, I am not sure how to compute an appropriate default for the cache size.

The eviction strategy, with the use of hashicorp/golang-lru, is to never grow beyond a fix size. However **in case the size limit is hit, the cache will evict out the LRU entry to accommodate the new entry.**. Thoughts ?

I have updated the perf numbers in the test file, numbers seem to be the same.